packages/ember-metal/lib/chains.ts

```diff
@@ -256,16 +248,14 @@ class ChainNode {
   }
 
   chain(key: string, tails: string[]): void {
-    let chains: { [key: string]: ChainNode | undefined };
-    if (this.chains === undefined) {
+    let chains = this.chains;
+    if (chains === undefined) {
       chains = this.chains = Object.create(null);
-    } else {
-      chains = this.chains;
     }
 
-    let node = chains[key];
+    let node = chains![key];
```

Comment:
I don't understand why the `!` is needed here.

Reply:
The `if (chains === undefined)` and assignment should remove undefined.

Reply:
Linter is complaining if I remove it

Reply:
probably bug in tslinter